src/common/Input/input.test.js

```diff
@@ -0,0 +1,159 @@
+import React from 'react'
+import { render, fireEvent, cleanup, screen } from '@testing-library/react'
+
+import Input from './Input'
+
+jest.mock('../../images/warning.svg', () => ({
+  ReactComponent: 'warning-icon'
+}))
+
+const component = props => render(<Input {...props} />)
+
+describe('Input component', () => {
+  let utils
+  const mockChange = jest.fn()
+
+  beforeEach(() => {
+    let props = {
+      className: 'input-default',
+      disabled: true,
+      floatingLabel: true,
+      iconClass: 'input-icon',
+      inputIcon: <>icon</>,
+      label: 'Name',
+      onChange: mockChange,
+      placeholder: 'Name',
+      required: false,
+      tip: 'This field is require',
+      type: 'text',
+      value: '123'
+    }
+    utils = component(props)
+  })
+
+  afterEach(cleanup)
+
+  it('renders without crashing', () => {
+    expect(utils.getByTestId('input')).not.toBeNull()
+  })
+
+  it('should props value set to "123"', () => {
+    expect(utils.getByTestId('input').value).toEqual('123')
+  })
+
+  it('should props placeholder set to "Name"', () => {
+    expect(utils.getByTestId('input').placeholder).toEqual('Name')
+  })
+
+  it('should props className set to "input-default"', () => {
+    expect(utils.getByTestId('input').className).toMatch('input-default')
+  })
+
+  it('should change value to "1243"', () => {
+    const value = '1243'
+    const input = utils.getByTestId('input')
+
+    fireEvent.change(input, { target: { value } })
+    expect(mockChange).toHaveBeenCalled()
+    utils.rerender(<Input type="text" value={value} />)
+    expect(input.value).toEqual('1243')
+  })
+
+  it('should calls the "onChange" callback handler', () => {
+    const input = utils.getByTestId('input')
+
+    fireEvent.change(input, { target: { value: '123' } })
+    expect(mockChange).toHaveBeenCalled()
+  })
+
+  it('should props disabled set to "true"', () => {
+    expect(utils.getByTestId('input').disabled).toEqual(true)
+  })
+
+  it('should show label if the props label is not empty', () => {
+    expect(utils.getByTestId('label')).not.toBeNull()
+  })
+
+  it('should input in focused if props focused set to "true"', () => {
+    cleanup()
+    const { getByTestId } = render(<Input type="text" focused={true} />)
+    const input = getByTestId('input')
+
+    expect(document.activeElement).toBe(input)
+  })
+
+  it('should show the label on the left in the middle of an input container if the input is empty', () => {
+    cleanup()
+    const props = {
+      type: 'text',
+      label: 'Name',
+      floatingLabel: true,
+      onChange: jest.fn()
+    }
+    const { getByTestId } = render(<Input {...props} />)
```

Comment:
you can use your renderComponentFunction and pass props directly. SO there is no need to declare const props
```suggestion
    const { getByTestId } = renderComponent({
      type: 'text',
      label: 'Name',
      floatingLabel: true,
      onChange: jest.fn()
    })
```